File: apps/web/src/features/elections/types/voter.ts
export interface VoterGroup {
  id: string;
  election_id: string;
  name: string;
  description: string | null;
  created_at: string;
  updated_at: string | null;
}

export interface EligibleVoter {
  id: string;
  election_id: string;
  voter_identifier: string;
  full_name: string;
  phone_number: string | null;
  email: string | null;
  gender: string | null;
  group_id: string | null;
  has_voted: boolean;
  created_at: string;
  updated_at: string | null;
}

export interface EligibleVoterCreate {
  voter_identifier: string;
  full_name: string;
  phone_number?: string | null;
  email?: string | null;
  gender?: string | null;
  group_id?: string | null;
}

export interface PaginatedVoterResponse {
  items: EligibleVoter[];
  total: number;
  skip: number;
  limit: number;
}

File: apps/api/app/modules/election/schemas/voter.py
import uuid
import datetime
from pydantic import BaseModel, ConfigDict, Field, EmailStr

class VoterGroupCreate(BaseModel):
    name: str = Field(..., max_length=150)
    description: str | None = Field(None, max_length=2000)

class VoterGroupResponse(BaseModel):
    id: uuid.UUID
    election_id: uuid.UUID
    name: str
    description: str | None
    created_at: datetime.datetime
    updated_at: datetime.datetime | None

    model_config = ConfigDict(from_attributes=True)

class EligibleVoterCreate(BaseModel):
    voter_identifier: str = Field(..., max_length=100)
    full_name: str = Field(..., max_length=255)
    phone_number: str | None = Field(None, max_length=50)
    email: EmailStr | None = None
    gender: str | None = Field(None, max_length=15)
    group_id: uuid.UUID | None = None

class EligibleVoterUpdate(BaseModel):
    full_name: str | None = Field(None, max_length=255)
    phone_number: str | None = Field(None, max_length=50)
    email: EmailStr | None = None
    gender: str | None = Field(None, max_length=15)
    group_id: uuid.UUID | None = None

class EligibleVoterResponse(BaseModel):
    id: uuid.UUID
    election_id: uuid.UUID
    voter_identifier: str
    full_name: str
    phone_number: str | None
    email: str | None
    gender: str | None
    group_id: uuid.UUID | None
    has_voted: bool
    created_at: datetime.datetime
    updated_at: datetime.datetime | None

    model_config = ConfigDict(from_attributes=True)

class PaginatedVoterResponse(BaseModel):
    items: list[EligibleVoterResponse]
    total: int
    skip: int
    limit: int
